File: jroner.com/public_html/cgi-bin/hw2/php/state-set.php
<?php
// /cgi-bin/hw2/php/state-set.php
session_name("CGISESSID");
session_save_path("/tmp/hw2_sessions");
@mkdir("/tmp/hw2_sessions", 0700, true);
session_start();

header("Cache-Control: no-cache");
header("Content-Type: text/html; charset=utf-8");

$sessId = session_id();
?>
<!DOCTYPE html>
<html>
<head><meta charset="utf-8"><title>PHP CGI Form</title></head>
<body>
  <h1>PHP CGI Form</h1>

  <p><b>Current Session ID:</b> <?= htmlspecialchars($sessId ?: "(none)") ?></p>

  <p>Enter a username and submit it to the PHP session page.</p>

  <form action="/cgi-bin/hw2/php/state.php" method="get">
    <label for="username">Username:</label>
    <input id="username" name="username" type="text" />
    <button type="submit">Save to Session</button>
  </form>

  <hr/>

  <ul>
    <li><a href="/cgi-bin/hw2/php/state.php">PHP Session</a></li>
    <li><a href="/cgi-bin/hw2/php/state-clear.php">Destroy Session</a></li>
  </ul>
</body>
</html>

File: jroner.com/public_html/cgi-bin/hw2/go/state-set.go
package main

import (
	"fmt"
	"net/http"
	"net/http/cgi"
	"html"
)

func handler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Cache-Control", "no-cache")
	w.Header().Set("Content-Type", "text/html; charset=utf-8")

	sess := ""
	if c, err := r.Cookie("CGISESSID"); err == nil {
		sess = c.Value
	}

	fmt.Fprint(w, "<!DOCTYPE html><html><head><meta charset='utf-8'><title>Go CGI Form</title></head><body>")
	fmt.Fprint(w, "<h1>Go CGI Form</h1>")
	if sess != "" {
		fmt.Fprintf(w, "<p><b>Current Session ID:</b> %s</p>", html.EscapeString(sess))
	} else {
		fmt.Fprint(w, "<p><b>Current Session ID:</b> (none yet)</p>")
	}

	fmt.Fprint(w, `
	<p>Enter a username and submit it to the Go session page.</p>
	<form action="/cgi-bin/hw2/go/state" method="get">
	  <label for="username">Username:</label>
	  <input id="username" name="username" type="text" />
	  <button type="submit">Save to Session</button>
	</form>
	<hr/>
	<ul>
	  <li><a href="/cgi-bin/hw2/go/state">Go Session</a></li>
	  <li><a href="/cgi-bin/hw2/go/state-clear">Destroy Session</a></li>
	</ul>
	</body></html>`)
}

func main() { cgi.Serve(http.HandlerFunc(handler)) }
